java/src/main/java/com/microsoft/graph/bulk/impl/ObjectToEdge.java

```diff
@@ -0,0 +1,281 @@
+        for (Method method : MethodUtils.getMethodsWithAnnotation(clazz, GremlinLabelGetter.class)) {
+            if (isStatic(method.getModifiers()) || !method.canAccess(from))
+                continue; // method is not accessible, attempts to call it will fail.
+            var edgeLabel = (String) method.invoke(from);
+
+            if (!edgeLabel.isBlank()) {
+                results.setLabel(edgeLabel);
+            }
+        }
+    }
+
+    /**
+     * Iterates throw the fields defined on the class and extracts values from fields marked with the
+     * appropriate annotations
+     *
+     * @param clazz   the class of the object being used
+     * @param results the GremlinEdge being updated
+     * @param from    the instance of the class to extract the values from
+     */
+    @SneakyThrows
+    private static void setFieldValues(Class<?> clazz, GremlinEdge results, Object from) {
+        for (Field field : FieldUtils.getAllFields(clazz)) {
+            if (isStatic(field.getModifiers()) || !field.canAccess(from)) continue; // Field is not accessible
+            setIdValue(field, results, from);
+            if (results.getLabel() == null) {
+                setLabel(field, results, from);
+            }
+            setSourceVertexValues(clazz, field, results, from);
+            setDestinationVertexValues(field, results, from);
+            setPropertyMap(field, results, from);
+            setPropertyValues(field, results, from);
+        }
+    }
+
+    /**
+     * Sets the value of the label if the field has been marked with a @GremlinLabel annotation.
+     * Will override any value already set for the label by either the class annotation or the getter method
+     *
+     * @param field   the field definition to extract the value using
+     * @param results the GremlinEdge being updated
+     * @param from    the instance of the class to extract the values from
+     * @throws IllegalAccessException When the field is not accessible from this utility
+     */
+    private static void setLabel(Field field, GremlinEdge results, Object from) throws IllegalAccessException {
+        if (!field.isAnnotationPresent(GremlinLabel.class)) return;
+
+        results.setLabel((String) field.get(from));
+    }
+
+    /**
+     * Sets the id value if the field has been marked with the GremlinId annotation
+     *
+     * @param field   the field definition to extract the value using
+     * @param results the GremlinEdge being updated
+     * @param from    the instance of the class to extract the values from
+     * @throws IllegalAccessException When the field is not accessible from this utility
+     */
+    private static void setIdValue(Field field, GremlinEdge results, Object from)
+            throws IllegalAccessException {
+        if (!field.isAnnotationPresent(GremlinId.class)) return;
+
+        results.setId((String) field.get(from));
+
+        if (results.getId() == null) {
```

Comment:
should we also check empty/blank string? is empty string allowed for id?

Reply:
Shouldn't be, I'll add that in there.